Cache/IO.lean

```diff
@@ -334,13 +346,12 @@ def unpackCache (hashMap : HashMap) (force : Bool) : IO Unit := do
     let args := (if force then #["-f"] else #[]) ++ #["-x", "-j", "-"]
     let child ← IO.Process.spawn { cmd := ← getLeanTar, args, stdin := .piped }
     let (stdin, child) ← child.takeStdin
-    let mathlibDepPath := (← mathlibDepPath).toString
     let config : Array Lean.Json := hashMap.fold (init := #[]) fun config path hash =>
       let pathStr := s!"{CACHEDIR / hash.asLTar}"
-      if isMathlibRoot || !isPathFromMathlib path then
-        config.push <| .str pathStr
-      else -- only mathlib files, when not in the mathlib4 repo, need to be redirected
-        config.push <| .mkObj [("file", pathStr), ("base", mathlibDepPath)]
+      if let some base := findRoot path then
+        config.push <| .mkObj [("file", pathStr), ("base", toString base)]
```

Comment:
@digama: this doesn't seem to be working properly for Std, but I can't see what LeanTar is doing with my `base` argument.

Can you teach it a verbose mode that prints out where the extracted files are being written to?

Reply:
The `else config` looks suspicious, that means to skip the file entirely.

The way this works is that you pass a json argument containing either `pathStr` or `{"file": pathStr, "base": base}`, and `base`, if provided, overrides the `baseDir` which is passed as the `-C` argument (with default value `.`). This says to decompress `pathStr` which is an ltar file, with `base` as the base for all generated files. The ltar file contains a list of relative paths and these are joined to `base`.

For the usual ltars generated by CI, the paths will be of the form:
* `.lake/build/lib/Mathlib/Tactic.trace` (and `.lake/build/lib/Mathlib/Tactic.{olean,ilean,c}[.hash]`) for files in mathlib
* `.lake/build/lib/proofwidgets/.lake/build/lib/ProofWidgets/Compat.trace` (etc) for files not in mathlib

Reply:
yes. `.lake/packages/proofwidgets/.lake/build/lib/ProofWidgets/Compat.trace` more specifically. (A cheap way to find out without additional tooling is `strings NNN.ltar | grep proofwidgets`)